src/main/java/de/tum/in/www1/artemis/service/connectors/GitService.java

```diff
@@ -621,22 +621,48 @@ public PullResult pull(Repository repo) throws GitAPIException {
     }
 
     /**
-     * Hard reset local repository to origin/master.
+     * Get branch that origin/HEAD points to, useful to handle default branches that are not master
      *
      * @param repo Local Repository Object.
+     * @return name of the origin/HEAD branch, e.g. 'main' or null if there is no HEAD
      */
-    public void resetToOriginMaster(Repository repo) {
+    public String getOriginHead(Repository repo) throws GitAPIException {
+        Git git = new Git(repo);
+        var originHeadRef = git.lsRemote().callAsMap().get("HEAD");
+        git.close();
+
+        // Empty Git repos don't have HEAD
+        if (originHeadRef == null) {
+            return null;
+        }
+
+        var fullName = originHeadRef.getTarget().getName();
+        return StringUtils.substringAfterLast(fullName, "/");
+    }
+
+    /**
+     * Hard reset local repository to origin/HEAD.
+     *
+     * @param repo Local Repository Object.
+     */
+    public void resetToOriginHead(Repository repo) {
         try {
             fetchAll(repo);
-            reset(repo, "origin/master");
+            var originHead = getOriginHead(repo);
+
+            if (originHead == null) {
+                log.error("Cannot hard reset the repo " + repo.getLocalPath() + " to origin/HEAD because it is empty.");
```

Comment:
I guess we should leave the method in this case, because the reset below won't work

Reply:
Good catch 👍